app/src/main/java/com/omkarmoghe/pokemap/MainActivity.java

```diff
@@ -17,30 +14,26 @@
 import com.omkarmoghe.pokemap.login.RequestCredentialsDialogFragment;
 import com.omkarmoghe.pokemap.map.MapWrapperFragment;
 import com.omkarmoghe.pokemap.settings.SettingsActivity;
+import com.omkarmoghe.pokemap.app_preferences.PokemapAppPreferences;
+import com.omkarmoghe.pokemap.app_preferences.PokemapSharedPreferences;
 
 public class MainActivity extends BaseActivity {
+    private static final String TAG = "Pokemap";
 
-    public static final String TAG = "Pokemap";
-
-    // fragments
-    private MapWrapperFragment mMapWrapperFragment;
-
-    // Preferences
-    SharedPreferences pref;
+    private PokemapAppPreferences pref;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        pref = PreferenceManager.getDefaultSharedPreferences(this);
+        pref = new PokemapSharedPreferences(this);
 
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
-        mMapWrapperFragment = MapWrapperFragment.newInstance();
         FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
-        transaction.replace(R.id.main_container, mMapWrapperFragment)
+        transaction.replace(R.id.main_container, MapWrapperFragment.newInstance())
```

Comment:
The new fragment was being created every time anyway so I didn't want to change that behaviour just in case it broke someone else's stuff.